contrib/platform/src/com/sun/jna/platform/win32/Netapi32.java

```diff
@@ -42,6 +44,82 @@
+     * @param UncClientName
+     *            Pointer to a string that specifies the name of the computer
+     *            session for which information is to be returned. If this parameter
+     *            is NULL, NetSessionEnum returns information for all computer
+     *            sessions on the server.
+     * @param username
+     *            Pointer to a string that specifies the name of the user for which
+     *            information is to be returned. If this parameter is NULL,
+     *            NetSessionEnum returns information for all users.
+     * @param level
+     *            Specifies the information level of the data. This parameter can be
+     *            one of 0, 1, 2, 10, 502.
+     * @param bufptr
+     *            Pointer to the buffer that receives the data. The format of this
+     *            data depends on the value of the level parameter, for example
+     *            {@code SESSION_INFO_0} for level 0.
+     *            <p>
+     *            This buffer is allocated by the system and must be freed using the
+     *            {@link #NetApiBufferFree} function. Note that you must free the
+     *            buffer even if the function fails with {@code ERROR_MORE_DATA}.
+     * @param prefmaxlen
+     *            Specifies the preferred maximum length of returned data, in bytes.
+     *            If you specify {@link #MAX_PREFERRED_LENGTH}, the function
+     *            allocates the amount of memory required for the data. If you
+     *            specify another value in this parameter, it can restrict the
+     *            number of bytes that the function returns. If the buffer size is
+     *            insufficient to hold all entries, the function returns
+     *            {@code ERROR_MORE_DATA}.
+     * @param entriesread
+     *            Pointer to a value that receives the count of elements actually
+     *            enumerated.
+     * @param totalentries
+     *            Pointer to a value that receives the total number of entries that
+     *            could have been enumerated from the current resume position. Note
+     *            that applications should consider this value only as a hint.
+     * @param resume_handle
+     *            Pointer to a value that contains a resume handle which is used to
+     *            continue an existing session search. The handle should be zero on
+     *            the first call and left unchanged for subsequent calls. If
+     *            resume_handle is NULL, no resume handle is stored.
+     * @return If the function succeeds, the return value is NERR_Success. If the
+     *         function fails, the return value is an error code.
+     */
+    int NetSessionEnum(String servername, String UncClientName, String username, int level, PointerByReference bufptr,
+            int prefmaxlen, IntByReference entriesread, IntByReference totalentries, IntByReference resume_handle);
```

Comment:
The string mapping might prove to be problematic, the headers are not consistent. See the differerence:

> NetGetJoinInformation(
    __in_opt IN   LPCWSTR                lpServer OPTIONAL,
    __out_opt OUT  LPWSTR                *lpNameBuffer,
    OUT  PNETSETUP_JOIN_STATUS  BufferType
    );

In this case the function call always expects a unicode string. The same is not true for:

> NetSessionEnum (
    __in_opt IN  LMSTR       servername OPTIONAL,
    __in_opt IN  LMSTR       UncClientName OPTIONAL,
    __in_opt IN  LMSTR       username OPTIONAL,
    IN  DWORD       level,
    __out OUT LPBYTE      *bufptr,
    IN  DWORD       prefmaxlen,
    __out OUT LPDWORD     entriesread,
    __out_opt OUT LPDWORD     totalentries,
    __inout_opt IN OUT LPDWORD  resume_handle OPTIONAL
    );

With the definition of `LMSTR` see comment in `SESSION_INFO_10` for the definition. And it depends on the interpretation there.

Reply:
Per above, `LMSTR` is always Unicode. So these should be `WString`.